src/components/main-view/main-view.jsx

```diff
@@ -21,35 +37,161 @@ export const MainView = () => {
+                {!user ? (
+                  <Navigate to="/login" replace />
+                ) : movies.length === 0 ? (
+                  <Col>The list is empty!</Col>
+                ) : (
+                  <Col md={8}>
+                    <MovieView movies={movies} />
+                  </Col>
+                )}
+              </>
+            }
+          />
+          <Route
+            path="/"
+            element={
+              <>
+                {!user ? (
+                  <Navigate to="/login" replace />
+                ) : movies.length === 0 ? (
+                  <Col>The list is empty</Col>
+                ) : (
+                  <>
+                    {movies.map((movie) => (
+                      <Col
+                        className="mb-4 container"
+                        key={movie.id}
+                        xs={12}
+                        sm={6}
+                        md={4}
+                        lg={3}
+                      >
+                        <MovieCard
+                          movie={movie}
+                          user={user}
+                          method={"POST"}
+                          messageOK={"Movie added"}
+                          messageFailed={"Failed to add movie"}
+                          fav={"Add to fav"}
+                        />
+                      </Col>
+                    ))}
+                  </>
+                )}
+              </>
+            }
+          />
+          <Route
+            path="/profile"
+            element={
+              <ProfileView
+                user={user}
+                onLoggedOut={hanldeLogout}
+                movies={movies}
+              />
+            }
+          />
+        </Routes>
+      </Row>
+    </BrowserRouter>
   );
 };
+
+// (
+//   <Container>
+//     <NavigationBar user={user} />
+//     <Row className="justify-content-md-center mt-4">
+//       {!user ? (
+//         <Col md={5}>
+//           <LoginView
+//             onLoggedIn={(user, token) => {
+//               setUser(user);
+//               setToken(token);
+//             }}
+//           />
+//           or
+//           <SignupView />
+//         </Col>
+//       ) : selectedMovie ? (
+//         <Col md={8}>
+//           <MovieView
+//             movie={selectedMovie}
+//             onBackClicked={() => setSelectedMovie(null)}
+//           />
+//         </Col>
+//       ) : movies.length === 0 ? (
+//         <div>The list is empty.</div>
+//       ) : (
+//         <>
+//           {movies.map((movie) => (
+//             <Col
+//               className="mb-5 mt-4"
+//               key={movie.id}
+//               xs={12}
+//               sm={6}
+//               md={4}
+//               lg={3}
+//             >
+//               <MovieCard
+//                 movie={movie}
+//                 onMovieClicked={(newSelectedMovie) => {
+//                   setSelectedMovie(newSelectedMovie);
+//                 }}
```

Comment:
B. After implementing A. I can pass the two functions to the MovieCard card component and I can pass the user object as props. The user object can be used to determine if a movie belong to the list of favs. The two function will be used to delete or add favs